Note, there is a strange ordering issue, when `verifyLiveTableUserResults` is executed second, the stack below is returned when calling `/bin/get/XWiki/UserDirectoryLivetableResults`

```
Failed to execute the [velocity] macro. Cause: [Index 0 out of bounds for length 0]. Click on this message for details.

org.xwiki.rendering.macro.MacroExecutionException: Failed to evaluate Velocity Macro for content [##
## This page is overridden in subwikis by UserDirectoryLivetableResultsOverride, which includes this page and extends
## the live table query to support global/local user filtering.
##
## Hide inactive user profiles if requested.
#if ($request.hideDisabledProfiles == 'true')
  #set ($extraFromClause = "$!extraFromClause, IntegerProperty as activeProperty")
  #set ($isActive = "obj.id = activeProperty.id.id and activeProperty.id.name = 'active' and " +
    'activeProperty.value = 1')
  #set ($extraWhereClause = "$!extraWhereClause and $isActive")
#end
#if ("$!extraWhereClauseParameters" == '')
  #set ($extraWhereClauseParameters = {})
#end
#set ($columns = $request.collist.split(','))
## Generate the live table results as JSON.
#gridresultwithfilter('XWiki.XWikiUsers' $columns "$!extraFromClause" "$!extraWhereClause" $extraWhereClauseParameters)]
 at org.xwiki.rendering.internal.macro.velocity.VelocityMacro.evaluateString(VelocityMacro.java:186)
 at org.xwiki.rendering.internal.macro.velocity.VelocityMacro.evaluateString(VelocityMacro.java:60)
 at org.xwiki.rendering.macro.script.AbstractScriptMacro.evaluateBlock(AbstractScriptMacro.java:279)
 at org.xwiki.rendering.macro.script.AbstractScriptMacro.execute(AbstractScriptMacro.java:182)
 at org.xwiki.rendering.macro.script.AbstractScriptMacro.execute(AbstractScriptMacro.java:58)
 at org.xwiki.rendering.internal.transformation.macro.MacroTransformation.transform(MacroTransformation.java:441)
 at org.xwiki.rendering.internal.transformation.DefaultRenderingContext.transformInContext(DefaultRenderingContext.java:183)
 at org.xwiki.rendering.internal.transformation.DefaultTransformationManager.performTransformations(DefaultTransformationManager.java:88)
 at org.xwiki.rendering.internal.transformation.XWikiTransformationManager.superPerformTransformations(XWikiTransformationManager.java:95)
 at org.xwiki.rendering.internal.transformation.XWikiTransformationManager.performTransformations(XWikiTransformationManager.java:88)
 at org.xwiki.display.internal.DocumentContentAsyncExecutor.executeInCurrentExecutionContext(DocumentContentAsyncExecutor.java:397)
 at org.xwiki.display.internal.DocumentContentAsyncExecutor.execute(DocumentContentAsyncExecutor.java:269)
 at org.xwiki.display.internal.DocumentContentAsyncRenderer.execute(DocumentContentAsyncRenderer.java:112)
 at org.xwiki.rendering.async.internal.block.AbstractBlockAsyncRenderer.render(AbstractBlockAsyncRenderer.java:157)
 at org.xwiki.rendering.async.internal.block.AbstractBlockAsyncRenderer.render(AbstractBlockAsyncRenderer.java:54)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.syncRender(DefaultAsyncRendererExecutor.java:290)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.render(DefaultAsyncRendererExecutor.java:267)
 at org.xwiki.rendering.async.internal.block.DefaultBlockAsyncRendererExecutor.execute(DefaultBlockAsyncRendererExecutor.java:125)
 at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:93)
 at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:46)
 at org.xwiki.display.internal.DefaultDocumentDisplayer.display(DefaultDocumentDisplayer.java:96)
 at org.xwiki.display.internal.DefaultDocumentDisplayer.display(DefaultDocumentDisplayer.java:39)
 at org.xwiki.sheet.internal.SheetDocumentDisplayer.display(SheetDocumentDisplayer.java:123)
 at org.xwiki.sheet.internal.SheetDocumentDisplayer.display(SheetDocumentDisplayer.java:52)
 at org.xwiki.display.internal.ConfiguredDocumentDisplayer.display(ConfiguredDocumentDisplayer.java:68)
 at org.xwiki.display.internal.ConfiguredDocumentDisplayer.display(ConfiguredDocumentDisplayer.java:42)
 at com.xpn.xwiki.doc.XWikiDocument.display(XWikiDocument.java:1433)
 at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1569)
 at com.xpn.xwiki.doc.XWikiDocument.displayDocument(XWikiDocument.java:1519)
 at com.xpn.xwiki.api.Document.displayDocument(Document.java:802)
 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
 at java.base/java.lang.reflect.Method.invoke(Method.java:580)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:571)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:554)
 at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:221)
 at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:368)
 at org.apache.velocity.runtime.parser.node.ASTReference.value(ASTReference.java:704)
 at org.apache.velocity.runtime.parser.node.ASTExpression.value(ASTExpression.java:75)
 at org.apache.velocity.runtime.parser.node.ASTSetDirective.render(ASTSetDirective.java:242)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:190)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:171)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.xwiki.velocity.internal.directive.TryCatchDirective.render(TryCatchDirective.java:86)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.Template.merge(Template.java:358)
 at org.apache.velocity.Template.merge(Template.java:262)
 at org.xwiki.velocity.internal.InternalVelocityEngine.evaluate(InternalVelocityEngine.java:233)
 at com.xpn.xwiki.internal.template.VelocityTemplateEvaluator.evaluateContent(VelocityTemplateEvaluator.java:107)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.evaluateContent(TemplateAsyncRenderer.java:219)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.renderVelocity(TemplateAsyncRenderer.java:174)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:135)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:54)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.lambda$syncRender$0(DefaultAsyncRendererExecutor.java:284)
 at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:98)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.syncRender(DefaultAsyncRendererExecutor.java:284)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.render(DefaultAsyncRendererExecutor.java:267)
 at org.xwiki.rendering.async.internal.block.DefaultBlockAsyncRendererExecutor.render(DefaultBlockAsyncRendererExecutor.java:154)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:869)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:831)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:818)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderNoException(InternalTemplateManager.java:773)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderNoException(InternalTemplateManager.java:765)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.renderNoException(DefaultTemplateManager.java:79)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.renderNoException(DefaultTemplateManager.java:73)
 at org.xwiki.template.script.TemplateScriptService.render(TemplateScriptService.java:54)
 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
 at java.base/java.lang.reflect.Method.invoke(Method.java:580)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:571)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:554)
 at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:221)
 at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:368)
 at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:492)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.Template.merge(Template.java:358)
 at org.apache.velocity.Template.merge(Template.java:262)
 at org.xwiki.velocity.internal.InternalVelocityEngine.evaluate(InternalVelocityEngine.java:233)
 at com.xpn.xwiki.internal.template.VelocityTemplateEvaluator.evaluateContent(VelocityTemplateEvaluator.java:107)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.evaluateContent(TemplateAsyncRenderer.java:219)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.renderVelocity(TemplateAsyncRenderer.java:174)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:135)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:54)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.lambda$syncRender$0(DefaultAsyncRendererExecutor.java:284)
 at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:98)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.syncRender(DefaultAsyncRendererExecutor.java:284)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.render(DefaultAsyncRendererExecutor.java:267)
 at org.xwiki.rendering.async.internal.block.DefaultBlockAsyncRendererExecutor.render(DefaultBlockAsyncRendererExecutor.java:154)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:869)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:831)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:818)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderNoException(InternalTemplateManager.java:773)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderNoException(InternalTemplateManager.java:765)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.renderNoException(DefaultTemplateManager.java:79)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.renderNoException(DefaultTemplateManager.java:73)
 at org.xwiki.template.script.TemplateScriptService.render(TemplateScriptService.java:54)
 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
 at java.base/java.lang.reflect.Method.invoke(Method.java:580)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:571)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:554)
 at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:221)
 at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:368)
 at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:492)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.Template.merge(Template.java:358)
 at org.apache.velocity.Template.merge(Template.java:262)
 at org.xwiki.velocity.internal.InternalVelocityEngine.evaluate(InternalVelocityEngine.java:233)
 at com.xpn.xwiki.internal.template.VelocityTemplateEvaluator.evaluateContent(VelocityTemplateEvaluator.java:107)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.evaluateContent(TemplateAsyncRenderer.java:219)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.renderVelocity(TemplateAsyncRenderer.java:174)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:135)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:54)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.lambda$syncRender$0(DefaultAsyncRendererExecutor.java:284)
 at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:98)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.syncRender(DefaultAsyncRendererExecutor.java:284)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.render(DefaultAsyncRendererExecutor.java:267)
 at org.xwiki.rendering.async.internal.block.DefaultBlockAsyncRendererExecutor.render(DefaultBlockAsyncRendererExecutor.java:154)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:869)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:831)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:818)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderNoException(InternalTemplateManager.java:773)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderNoException(InternalTemplateManager.java:765)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.renderNoException(DefaultTemplateManager.java:79)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.renderNoException(DefaultTemplateManager.java:73)
 at org.xwiki.template.script.TemplateScriptService.render(TemplateScriptService.java:54)
 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
 at java.base/java.lang.reflect.Method.invoke(Method.java:580)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:571)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:554)
 at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:221)
 at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:368)
 at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:492)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.Template.merge(Template.java:358)
 at org.apache.velocity.Template.merge(Template.java:262)
 at org.xwiki.velocity.internal.InternalVelocityEngine.evaluate(InternalVelocityEngine.java:233)
 at com.xpn.xwiki.internal.template.VelocityTemplateEvaluator.evaluateContent(VelocityTemplateEvaluator.java:107)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.evaluateContent(TemplateAsyncRenderer.java:219)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.renderVelocity(TemplateAsyncRenderer.java:174)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:135)
 at com.xpn.xwiki.internal.template.TemplateAsyncRenderer.render(TemplateAsyncRenderer.java:54)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.lambda$syncRender$0(DefaultAsyncRendererExecutor.java:284)
 at com.xpn.xwiki.internal.security.authorization.DefaultAuthorExecutor.call(DefaultAuthorExecutor.java:98)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.syncRender(DefaultAsyncRendererExecutor.java:284)
 at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.render(DefaultAsyncRendererExecutor.java:267)
 at org.xwiki.rendering.async.internal.block.DefaultBlockAsyncRendererExecutor.render(DefaultBlockAsyncRendererExecutor.java:154)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:869)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:831)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.renderFromSkin(InternalTemplateManager.java:811)
 at com.xpn.xwiki.internal.template.InternalTemplateManager.render(InternalTemplateManager.java:797)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.render(DefaultTemplateManager.java:91)
 at com.xpn.xwiki.internal.template.DefaultTemplateManager.render(DefaultTemplateManager.java:85)
 at com.xpn.xwiki.XWiki.evaluateTemplate(XWiki.java:2566)
 at com.xpn.xwiki.web.Utils.parseTemplate(Utils.java:180)
 at com.xpn.xwiki.web.XWikiAction.execute(XWikiAction.java:651)
 at com.xpn.xwiki.web.XWikiAction.execute(XWikiAction.java:339)
 at com.xpn.xwiki.web.LegacyActionServlet.service(LegacyActionServlet.java:111)
 at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:723)
 at org.eclipse.jetty.ee11.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1397)
 at org.eclipse.jetty.ee11.servlet.ServletHolder.handle(ServletHolder.java:754)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1668)
 at com.xpn.xwiki.web.ActionFilter.doFilter(ActionFilter.java:123)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.wysiwyg.filter.ConversionFilter.doFilter(ConversionFilter.java:71)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.container.servlet.filters.internal.SetHTTPHeaderFilter.doFilter(SetHTTPHeaderFilter.java:66)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.resource.servlet.RoutingFilter.doFilter(RoutingFilter.java:135)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.container.servlet.filters.internal.SavedRequestRestorerFilter.doFilter(SavedRequestRestorerFilter.java:212)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.container.servlet.filters.internal.SafeRedirectFilter.doFilter(SafeRedirectFilter.java:106)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.container.servlet.filters.internal.ResolveRelativeRedirectFilter.doFilter(ResolveRelativeRedirectFilter.java:129)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.container.servlet.filters.internal.SourceURLResolverFilter.doFilter(SourceURLResolverFilter.java:177)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.xwiki.container.servlet.filters.internal.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:120)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:208)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.eclipse.jetty.ee11.websocket.servlet.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:199)
 at org.eclipse.jetty.ee11.servlet.FilterHolder.doFilter(FilterHolder.java:205)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1640)
 at org.eclipse.jetty.ee11.servlet.ServletHandler$MappedServlet.handle(ServletHandler.java:1602)
 at org.eclipse.jetty.ee11.servlet.ServletChannel.dispatch(ServletChannel.java:871)
 at org.eclipse.jetty.ee11.servlet.ServletChannel.handle(ServletChannel.java:449)
 at org.eclipse.jetty.ee11.servlet.ServletHandler.handle(ServletHandler.java:469)
 at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:546)
 at org.eclipse.jetty.ee11.servlet.SessionHandler.handle(SessionHandler.java:763)
 at org.eclipse.jetty.server.handler.ContextHandler.handle(ContextHandler.java:1250)
 at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:148)
 at org.eclipse.jetty.server.Server.handle(Server.java:197)
 at org.eclipse.jetty.server.internal.HttpChannelState$HandlerInvoker.run(HttpChannelState.java:793)
 at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:419)
 at org.eclipse.jetty.server.internal.HttpConnection$FillableCallback.succeeded(HttpConnection.java:1780)
 at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
 at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:54)
 at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:492)
 at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.epcRunTask(AdaptiveExecutionStrategy.java:428)
 at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:401)
 at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:255)
 at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:204)
 at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:317)
 at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:1009)
 at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1240)
 at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1194)
 at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.xwiki.velocity.XWikiVelocityException: Failed to evaluate content with namespace [xwiki:XWiki.UserDirectoryLivetableResults]
 at org.xwiki.velocity.internal.InternalVelocityEngine.evaluate(InternalVelocityEngine.java:235)
 at org.xwiki.rendering.internal.macro.velocity.VelocityMacro.evaluateString(VelocityMacro.java:168)
 ... 231 more
Caused by: org.apache.velocity.exception.MethodInvocationException: Invocation of method 'getObfuscatedValue' in  class com.xpn.xwiki.api.Property threw exception java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0 at xwiki:XWiki.LiveTableResultsMacros[line 468, column 37]
 at org.apache.velocity.runtime.parser.node.ASTMethod.handleInvocationException(ASTMethod.java:308)
 at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:235)
 at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:368)
 at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:492)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.runtime.parser.node.ASTStringLiteral.value(ASTStringLiteral.java:320)
 at org.apache.velocity.runtime.parser.node.ASTExpression.value(ASTExpression.java:75)
 at org.apache.velocity.runtime.parser.node.ASTSetDirective.render(ASTSetDirective.java:242)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:190)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.Foreach.renderBlock(Foreach.java:309)
 at org.apache.velocity.runtime.directive.Foreach.render(Foreach.java:279)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:171)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:171)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.Foreach.renderBlock(Foreach.java:309)
 at org.apache.velocity.runtime.directive.Foreach.render(Foreach.java:279)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.parser.node.ASTIfStatement.render(ASTIfStatement.java:171)
 at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:147)
 at org.apache.velocity.runtime.directive.VelocimacroProxy.render(VelocimacroProxy.java:217)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:331)
 at org.apache.velocity.runtime.directive.RuntimeMacro.render(RuntimeMacro.java:261)
 at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:304)
 at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:439)
 at org.apache.velocity.Template.merge(Template.java:358)
 at org.apache.velocity.Template.merge(Template.java:262)
 at org.xwiki.velocity.internal.InternalVelocityEngine.evaluate(InternalVelocityEngine.java:233)
 ... 232 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
 at com.xpn.xwiki.objects.classes.EmailClass.getObfuscatedValue(EmailClass.java:114)
 at com.xpn.xwiki.objects.BaseProperty.getObfuscatedValue(BaseProperty.java:240)
 at com.xpn.xwiki.objects.BaseStringProperty.getObfuscatedValue(BaseStringProperty.java:54)
 at com.xpn.xwiki.api.Property.getObfuscatedValue(Property.java:108)
 at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
 at java.base/java.lang.reflect.Method.invoke(Method.java:580)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:571)
 at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:554)
 at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:221)
 ... 284 more
```

--- a/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/EmailClass.java
+++ b/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/EmailClass.java
@@ -111,8 +111,13 @@ public class EmailClass extends StringClass
     {
         EmailAddressObfuscator emailAddressObfuscator = Utils.getComponent(EmailAddressObfuscator.class);
         try {
-            InternetAddress address = InternetAddress.parse(String.valueOf(value))[0];
-            return emailAddressObfuscator.obfuscate(address);
+            // InternetAddress.parse returns an empty array (rather than throwing) when the value is blank, so we
+            // can't blindly access index 0.
+            InternetAddress[] addresses = InternetAddress.parse(String.valueOf(value));
+            if (addresses.length == 0) {
+                return null;
+            }
+            return emailAddressObfuscator.obfuscate(addresses[0]);
         } catch (AddressException e) {
             LOGGER.debug("Invalid email address value when trying to obfuscate [{}] falling back on null.", value);
             return null;

--- a/xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/objects/classes/EmailClassTest.java
+++ b/xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/objects/classes/EmailClassTest.java
@@ -1,0 +1,89 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xpn.xwiki.objects.classes;
+
+import javax.mail.internet.InternetAddress;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.xwiki.mail.EmailAddressObfuscator;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectComponentManager;
+import org.xwiki.test.mockito.MockitoComponentManager;
+
+import com.xpn.xwiki.web.Utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link EmailClass}.
+ *
+ * @version $Id$
+ */
+@ComponentTest
+class EmailClassTest
+{
+    @InjectComponentManager
+    private MockitoComponentManager componentManager;
+
+    private EmailAddressObfuscator emailAddressObfuscator;
+
+    private EmailClass emailClass;
+
+    @BeforeEach
+    void setUp() throws Exception
+    {
+        Utils.setComponentManager(this.componentManager);
+        this.emailAddressObfuscator = this.componentManager.registerMockComponent(EmailAddressObfuscator.class);
+        this.emailClass = new EmailClass();
+    }
+
+    @Test
+    void getObfuscatedValueWithValidEmail()
+    {
+        when(this.emailAddressObfuscator.obfuscate(
+            argThat(address -> "john@doe.com".equals(address.getAddress())))).thenReturn("j...@doe.com");
+
+        assertEquals("j...@doe.com", this.emailClass.getObfuscatedValue("john@doe.com"));
+    }
+
+    @Test
+    void getObfuscatedValueWithEmptyString()
+    {
+        // InternetAddress.parse("") returns an empty array (rather than throwing AddressException). The implementation
+        // must handle this without raising ArrayIndexOutOfBoundsException.
+        assertNull(this.emailClass.getObfuscatedValue(""));
+    }
+
+    @Test
+    void getObfuscatedValueWithNullValue()
+    {
+        assertNull(this.emailClass.getObfuscatedValue(null));
+    }
+
+    @Test
+    void getObfuscatedValueWithMalformedAddress()
+    {
+        assertNull(this.emailClass.getObfuscatedValue("not a valid email"));
+    }
+}

--- a/xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/objects/classes/EmailClassTest.java
+++ b/xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/objects/classes/EmailClassTest.java
@@ -19,6 +19,7 @@
  */
 package com.xpn.xwiki.objects.classes;
 
+import javax.mail.internet.AddressException;
 import javax.mail.internet.InternetAddress;
 
 import org.junit.jupiter.api.BeforeEach;
@@ -32,7 +33,6 @@ import com.xpn.xwiki.web.Utils;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.mockito.ArgumentMatchers.argThat;
 import static org.mockito.Mockito.when;
 
 /**
@@ -48,22 +48,20 @@ class EmailClassTest
 
     private EmailAddressObfuscator emailAddressObfuscator;
 
-    private EmailClass emailClass;
+    private final EmailClass emailClass = new EmailClass();
 
     @BeforeEach
     void setUp() throws Exception
     {
         Utils.setComponentManager(this.componentManager);
         this.emailAddressObfuscator = this.componentManager.registerMockComponent(EmailAddressObfuscator.class);
-        this.emailClass = new EmailClass();
     }
 
     @Test
-    void getObfuscatedValueWithValidEmail()
+    void getObfuscatedValueWithValidEmail() throws AddressException
     {
-        when(this.emailAddressObfuscator.obfuscate(
-            argThat(address -> "john@doe.com".equals(address.getAddress())))).thenReturn("j...@doe.com");
-
+        when(this.emailAddressObfuscator.obfuscate(InternetAddress.parse("john@doe.com")[0]))
+            .thenReturn("j...@doe.com");
         assertEquals("j...@doe.com", this.emailClass.getObfuscatedValue("john@doe.com"));
     }
 

--- a/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-directory/xwiki-platform-user-directory-test/xwiki-platform-user-directory-test-docker/src/test/it/org/xwiki/user/directory/test/ui/UserDirectoryIT.java
+++ b/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-directory/xwiki-platform-user-directory-test/xwiki-platform-user-directory-test-docker/src/test/it/org/xwiki/user/directory/test/ui/UserDirectoryIT.java
@@ -21,6 +21,7 @@ package org.xwiki.user.directory.test.ui;
 
 import java.util.Map;
 
+import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.xwiki.test.docker.junit5.UITest;
 import org.xwiki.test.integration.junit.LogCaptureConfiguration;
@@ -68,12 +69,44 @@ import static org.junit.jupiter.api.Assertions.assertTrue;
 class UserDirectoryIT
 {
     @Test
+    @Order(1)
+    void verifyLiveTableUserResults(TestUtils testUtils) throws Exception
+    {
+        testUtils.loginAsSuperAdmin();
+        testUtils.createUser("UserDirectoryITUser", "foobar", null, "email", "foo@xwiki.com");
+        testUtils.forceGuestUser();
+
+        String jsonResult = testUtils.getString("/bin/get/XWiki/UserDirectoryLivetableResults", Map.of(
+            "outputSyntax", "plain",
+            "transprefix", "xe.userdirectory.",
+            "classname", "XWiki.XWikiUsers",
+            "collist", "doc.name,first_name,last_name,email,password",
+            "sort", "doc.creationDate"
+        ));
+        System.out.println(jsonResult);
+        JsonNode jsonNode = new ObjectMapper().readTree(jsonResult);
+        assertInstanceOf(ObjectNode.class, jsonNode);
+        ObjectNode objectNode = (ObjectNode) jsonNode;
+        ArrayNode rows = (ArrayNode) objectNode.get("rows");
+        assertEquals(1, rows.size());
+
+        ObjectNode userCreated = (ObjectNode) rows.get(0);
+        assertEquals("XWiki.UserDirectoryITUser", userCreated.get("doc_fullName").asText());
+        assertEquals("********",  userCreated.get("password").asText().trim());
+        assertEquals("",  userCreated.get("password_value").asText().trim());
+        assertEquals("f...@xwiki.com",  userCreated.get("email").asText().trim());
+        assertEquals("f...@xwiki.com",  userCreated.get("email_value").asText().trim());
+    }
+
+    @Test
+    @Order(2)
     void verifyUserIsListed(TestUtils setup, LogCaptureConfiguration logCaptureConfiguration)
     {
         setup.loginAsSuperAdmin();
 
-        // Delete possible existing user
+        // Delete possible existing users
         setup.deletePage("XWiki", "test");
+        setup.deletePage("XWiki", "UserDirectoryITUser");
 
         UserDirectoryPage page = UserDirectoryPage.gotoPage();
 
@@ -106,33 +139,5 @@ class UserDirectoryIT
             "Exception in macro #displayCheckedIfWatched called at",
             "Exception in macro #generateNotificationInput called at"
         );
-    }
-
-    @Test
-    void verifyLiveTableUserResults(TestUtils testUtils) throws Exception
-    {
-        testUtils.loginAsSuperAdmin();
-        testUtils.createUser("UserDirectoryITUser", "foobar", null, "email", "foo@xwiki.com");
-        testUtils.forceGuestUser();
-
-        String jsonResult = testUtils.getString("/bin/get/XWiki/UserDirectoryLivetableResults", Map.of(
-            "outputSyntax", "plain",
-            "transprefix", "xe.userdirectory.",
-            "classname", "XWiki.XWikiUsers",
-            "collist", "doc.name,first_name,last_name,email,password",
-            "sort", "doc.creationDate"
-        ));
-        JsonNode jsonNode = new ObjectMapper().readTree(jsonResult);
-        assertInstanceOf(ObjectNode.class, jsonNode);
-        ObjectNode objectNode = (ObjectNode) jsonNode;
-        ArrayNode rows = (ArrayNode) objectNode.get("rows");
-        assertEquals(1, rows.size());
-
-        ObjectNode userCreated = (ObjectNode) rows.get(0);
-        assertEquals("XWiki.UserDirectoryITUser", userCreated.get("doc_fullName").asText());
-        assertEquals("********",  userCreated.get("password").asText().trim());
-        assertEquals("",  userCreated.get("password_value").asText().trim());
-        assertEquals("f...@xwiki.com",  userCreated.get("email").asText().trim());
-        assertEquals("f...@xwiki.com",  userCreated.get("email_value").asText().trim());
     }
 }

--- a/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-directory/xwiki-platform-user-directory-test/xwiki-platform-user-directory-test-docker/src/test/it/org/xwiki/user/directory/test/ui/UserDirectoryIT.java
+++ b/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-directory/xwiki-platform-user-directory-test/xwiki-platform-user-directory-test-docker/src/test/it/org/xwiki/user/directory/test/ui/UserDirectoryIT.java
@@ -21,6 +21,7 @@ package org.xwiki.user.directory.test.ui;
 
 import java.util.Map;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.xwiki.test.docker.junit5.UITest;
@@ -68,45 +69,21 @@ import static org.junit.jupiter.api.Assertions.assertTrue;
 )
 class UserDirectoryIT
 {
-    @Test
-    @Order(1)
-    void verifyLiveTableUserResults(TestUtils testUtils) throws Exception
+    @BeforeEach
+    void setUp(TestUtils setup)
     {
-        testUtils.loginAsSuperAdmin();
-        testUtils.createUser("UserDirectoryITUser", "foobar", null, "email", "foo@xwiki.com");
-        testUtils.forceGuestUser();
-
-        String jsonResult = testUtils.getString("/bin/get/XWiki/UserDirectoryLivetableResults", Map.of(
-            "outputSyntax", "plain",
-            "transprefix", "xe.userdirectory.",
-            "classname", "XWiki.XWikiUsers",
-            "collist", "doc.name,first_name,last_name,email,password",
-            "sort", "doc.creationDate"
-        ));
-        System.out.println(jsonResult);
-        JsonNode jsonNode = new ObjectMapper().readTree(jsonResult);
-        assertInstanceOf(ObjectNode.class, jsonNode);
-        ObjectNode objectNode = (ObjectNode) jsonNode;
-        ArrayNode rows = (ArrayNode) objectNode.get("rows");
-        assertEquals(1, rows.size());
-
-        ObjectNode userCreated = (ObjectNode) rows.get(0);
-        assertEquals("XWiki.UserDirectoryITUser", userCreated.get("doc_fullName").asText());
-        assertEquals("********",  userCreated.get("password").asText().trim());
-        assertEquals("",  userCreated.get("password_value").asText().trim());
-        assertEquals("f...@xwiki.com",  userCreated.get("email").asText().trim());
-        assertEquals("f...@xwiki.com",  userCreated.get("email_value").asText().trim());
+        setup.loginAsSuperAdmin();
+        setup.deletePage("XWiki", "test");
+        setup.deletePage("XWiki", "UserDirectoryITUser");
     }
 
     @Test
-    @Order(2)
+    @Order(1)
     void verifyUserIsListed(TestUtils setup, LogCaptureConfiguration logCaptureConfiguration)
     {
         setup.loginAsSuperAdmin();
 
         // Delete possible existing users
-        setup.deletePage("XWiki", "test");
-        setup.deletePage("XWiki", "UserDirectoryITUser");
 
         UserDirectoryPage page = UserDirectoryPage.gotoPage();
 
@@ -139,5 +116,34 @@ class UserDirectoryIT
             "Exception in macro #displayCheckedIfWatched called at",
             "Exception in macro #generateNotificationInput called at"
         );
+    }
+
+    @Test
+    @Order(2)
+    void verifyLiveTableUserResults(TestUtils testUtils) throws Exception
+    {
+        testUtils.loginAsSuperAdmin();
+        testUtils.createUser("UserDirectoryITUser", "foobar", null, "email", "foo@xwiki.com");
+        testUtils.forceGuestUser();
+
+        String jsonResult = testUtils.getString("/bin/get/XWiki/UserDirectoryLivetableResults", Map.of(
+            "outputSyntax", "plain",
+            "transprefix", "xe.userdirectory.",
+            "classname", "XWiki.XWikiUsers",
+            "collist", "doc.name,first_name,last_name,email,password",
+            "sort", "doc.creationDate"
+        ));
+        JsonNode jsonNode = new ObjectMapper().readTree(jsonResult);
+        assertInstanceOf(ObjectNode.class, jsonNode);
+        ObjectNode objectNode = (ObjectNode) jsonNode;
+        ArrayNode rows = (ArrayNode) objectNode.get("rows");
+        assertEquals(1, rows.size());
+
+        ObjectNode userCreated = (ObjectNode) rows.get(0);
+        assertEquals("XWiki.UserDirectoryITUser", userCreated.get("doc_fullName").asText());
+        assertEquals("********",  userCreated.get("password").asText().trim());
+        assertEquals("",  userCreated.get("password_value").asText().trim());
+        assertEquals("f...@xwiki.com",  userCreated.get("email").asText().trim());
+        assertEquals("f...@xwiki.com",  userCreated.get("email_value").asText().trim());
     }
 }

